packages/neon-core/src/tx/transaction/main.ts

```diff
@@ -23,51 +25,87 @@ export function serializeArrayOf(prop: any[]): string {
   return num2VarInt(prop.length) + prop.map(p => p.serialize()).join("");
 }
 
-export function deserializeType(
+export function deserializeVersion(
   ss: StringStream,
   tx: Partial<TransactionLike> = {}
 ): Partial<TransactionLike> {
   const byte = ss.read();
-  return Object.assign(tx, { type: parseInt(byte, 16) });
+  const version = parseInt(byte, 16);
+  if (version !== 0) {
+    throw new Error(`Transaction version should be 0 not ${version}`);
+  }
+  return Object.assign(tx, { version });
 }
 
-export function deserializeVersion(
+export function deserializeNonce(
   ss: StringStream,
   tx: Partial<TransactionLike> = {}
 ): Partial<TransactionLike> {
-  const byte = ss.read();
-  return Object.assign({ version: parseInt(byte, 16) });
+  const nonce = parseInt(ss.read(4), 16);
+  return Object.assign(tx, { nonce });
 }
 
-export function deserializeAttributes(
+export function deserializeSender(
   ss: StringStream,
-  tx: Partial<TransactionLike>
+  tx: Partial<TransactionLike> = {}
 ): Partial<TransactionLike> {
-  const attributes = deserializeArrayOf(
-    TransactionAttribute.fromStream,
-    ss
-  ).map(i => i.export());
-  return Object.assign(tx, { attributes });
+  return Object.assign(tx, { sender: ss.read(20) });
 }
 
-export function deserializeInputs(
+export function deserializeScript(
+  ss: StringStream,
+  tx: Partial<TransactionLike> = {}
+): Partial<TransactionLike> {
+  const script = ss.readVarBytes();
+  if (script.length === 0) {
+    throw new Error("Script should not be vacant.");

```

Comment:
Unless this is an illegal state, we should not throw anything. Let it deserialize.

Reply:
https://github.com/neo-project/neo/blob/2ef5d34cd88facbdfd13fd81edd8ad25012879b3/neo/Network/P2P/Payloads/Transaction.cs#L98
The core code throws.
I think it's because a transaction cannot be vacant.

Reply:
The core code expects fully formed transactions. We are trying to construct it here so I think its fine to let it go empty.

Reply:
OK.

Reply:
This is still throwing an error.

Reply:
Sorry this must be a mistake I made when resolving conflicts. Changed